Improve docs and messages

nickolay opened the pull request:
See individual commits for details.

awdeorio commented:
Thanks for corrections, I'll take a more detailed look soon.

awdeorio commented on README.md:
```diff
@@ -5,9 +5,9 @@ Mailmerge
 [![codecov](https://codecov.io/gh/awdeorio/mailmerge/branch/develop/graph/badge.svg)](https://codecov.io/gh/awdeorio/mailmerge)
 
-A simple, command line mail merge tool.
+A simple, command line [mail merge](https://en.wikipedia.org/wiki/Mail_merge) tool.
```
I think I'd prefer to keep the original text on this one.

nickolay replied:
Sure, if you say so... I had no idea "mail merge" was a standard term for this, so this looked like a poor choice of words until I googled it (which was after I read all of the README).

nickolay replied:
Reverted to the original text, assuming that's what you meant with your thumbs up.

awdeorio commented on README.md:
```diff
@@ -5,9 +5,9 @@ Mailmerge
+A simple, command line [mail merge](https://en.wikipedia.org/wiki/Mail_merge) tool.
 
-`mailmerge` uses plain text files and the [jinja2 template engine](http://jinja.pocoo.org/docs/latest/templates/).
+`mailmerge` generates and sends e-mails based on a [jinja2](http://jinja.pocoo.org/docs/latest/templates/) template, filling it with data from a CSV file.
```
```suggestion
Write a template email using [jinja2](http://jinja.pocoo.org/docs/latest/templates/) syntax and `mailmerge` will fill it in from a CSV file.
```

nickolay replied:
I agree this is better. May I suggest mentioning the SMTP functionality explicitly: "[...will fill it in from a CSV file] and, optionally, send it"?

awdeorio replied:
Sure, sounds good

awdeorio commented on README.md:
```diff
@@ -107,7 +107,7 @@ Date: Thu, 19 Dec 2019 19:49:11 -0000
 
 Your number is 17.
->>> sent message 1
+>>> message 1 not sent (dry run)
```
I see what you're getting at here.  I don't want the user to be confused and think that a message was sent.  I also don't want them to be confused and think there was an error.  How about this?

```suggestion
>>> message 1 sent (dry run)
```

nickolay replied:
Well, to me _"sent (dry run)"_ sounds self-contradictory and thus more confusing.
"Would be sent, if not for --dry-run" or something to that effect would be OK.
Happy to change to whatever wording you prefer.

nickolay replied:
Not sure which wording you prefer here; kept the original for now.

awdeorio replied:
Having something that's consistent and parseable by other tools is a plus.  I like how similar `message 1` and `message 1 sent` are.  Let's stick with `>>> message 1 sent (dry run)`

awdeorio commented on mailmerge/__main__.py:
```diff
@@ -71,13 +71,13 @@
     default="colorized",
     type=click.Choice(["colorized", "text", "raw"]),
-    help="Output format (colorized).",
+    help="Logging output format (colorized).",
```
I would prefer the original.  There's a plan to add logging which is different.

nickolay replied:
From my perspective, *the e-mails* are the real output of the program, and I honestly didn't expect the console output to be configurable. I didn't know about the [csplit trick](https://github.com/awdeorio/mailmerge/pull/107#issuecomment-720706463) that allows one to collect the e-mails from stdout.
Maybe: "Console output format (colorized)"?

awdeorio replied:
"Console output format" sounds good

awdeorio commented on mailmerge/__main__.py:
```diff
@@ -71,13 +71,13 @@
     default="mailmerge_server.conf",
     type=click.Path(),
-    help="server configuration (mailmerge_server.conf)",
+    help="SMTP configuration (mailmerge_server.conf)",
```
```suggestion
    help="SMTP server configuration (mailmerge_server.conf)",
```

nickolay replied:
That makes the `--help` output wrap for me, which is why I picked the shorter version:
```
  --database PATH                 database CSV (mailmerge_database.csv)
  --config PATH                   SMTP server configuration
                                  (mailmerge_server.conf)

  --output-format [colorized|text|raw]
```

vs 
```
  --config PATH                   SMTP configuration (mailmerge_server.conf)
```

awdeorio replied:
sure that's fine

awdeorio commented on mailmerge/template_message.py:
```diff
@@ -70,8 +70,8 @@ def render(self, context):
         self._message.__setitem__('Date', email.utils.formatdate())
-        assert self._sender
-        assert self._recipients
+        assert self._sender, "Missing FROM header in the template"
```
I like this

awdeorio commented on mailmerge/template_message.py:
```diff
@@ -70,8 +70,8 @@ def render(self, context):
-        assert self._sender
-        assert self._recipients
+        assert self._sender, "Missing FROM header in the template"
+        assert self._recipients, "Missing TO/CC headers in the template"
```
Could also include `BCC`

nickolay replied:
Will change after other points are finalized, even though I omitted BCC on purpose.

awdeorio reviewed (changes requested):
Thanks for the detailed read-through of the README and the suggestions for the command line UI.  It's really helpful to have the perspective of a new user.

I left some comments after taking a closer look.

nickolay commented:
Thanks for the replies. I updated pushed a few fixup commits (can rebase whenever you prefer). I believe there's [only one unaddressed comment](https://github.com/awdeorio/mailmerge/pull/108#discussion_r516214212) left.
